packages/api/src/ai/services/memory.service.ts

```diff
@@ -70,4 +70,26 @@ export class MemoryService {
       })
     );
   }
+
+  async createMemoryWithDocumentInput(
+    roomId: string,
+    input: string,
+    response: string
+  ) {
+    const redisDocumentResponse = new RedisChatMessageHistory({
```

Comment:
```suggestion
    const redisChatHistory = new RedisChatMessageHistory({
```

Reply:
also, wouldn't it be better to fetch the memory in cache before creating a new one? just leverage the existing methods in memory service